src/table/selection/selection-cell.tsx

```diff
@@ -0,0 +1,64 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+
+import ScreenreaderOnly from '../../internal/components/screenreader-only';
+import { TableTdElement, TableTdElementProps } from '../body-cell/td-element';
+import { TableThElement, TableThElementProps } from '../header-cell/th-element';
+import { Divider } from '../resizer';
+import { SelectionProps } from './interfaces';
+import { SelectionControl, SelectionControlProps } from './selection-control';
+
+import styles from '../styles.css.js';
+
+interface TableHeaderSelectionCellProps extends Omit<TableThElementProps, 'children' | 'colIndex'> {
+  focusedComponent?: null | string;
+  singleSelectionHeaderAriaLabel?: string;
+  getSelectAllProps?: () => SelectionProps;
+  onFocusMove: ((sourceElement: HTMLElement, fromIndex: number, direction: -1 | 1) => void) | undefined;
+}
+
+interface TableBodySelectionCellProps
+  extends Omit<TableTdElementProps, 'children' | 'colIndex' | 'wrapLines' | 'isEditable' | 'isEditing'> {
+  selectionControlProps?: SelectionControlProps;
+}
+
+export function TableHeaderSelectionCell({
+  focusedComponent,
+  singleSelectionHeaderAriaLabel,
+  getSelectAllProps,
+  onFocusMove,
+  ...props
+}: TableHeaderSelectionCellProps) {
+  return (
+    <TableThElement
+      {...props}
+      hidden={props.hidden}
```

Comment:
Wouldn't this be included with `{...props}` anyway?

Reply:
Yes, will remove